### PR TITLE
crypto: add StringCompressed() for PublicKey

### LINKED ISSUE #3263: Compressed stringer for public keys
## Is your feature request related to a problem? Please describe.
I'm always frustrated when I realize people have to do `hex.EncodeToString(key.Bytes())` to get a string representation of the key in compressed form. They have https://pkg.go.dev/github.com/nspcc-dev/neo-go/pkg/crypto/keys#NewPublicKeyFromString to go the other way around, but they can't easily create this string.

## Describe the solution you'd like
Some additional method to have a compressed hex string of the public key.

## Describe alternatives you've considered
`String()` method returns uncompressed key and likely can't be changed at this point.

### PR DESCRIPTION
**Pull request by AliceInHunterland**
Close #3263

### CODE REVIEW

File: pkg/crypto/keys/publickey.go [hunk 1]
```diff
@@ -406,3 +406,9 @@ func (p *PublicKey) UnmarshalYAML(unmarshal func(any) error) error {
 	}
 	return p.DecodeBytes(b)
 }
+
+// CompressedHexString returns the hex string representation of the public key
+// in its compressed form.
+func (p *PublicKey) CompressedHexString() string {
```
Comment (AnnaShaleva) [inline pkg/crypto/keys/publickey.go:412]: I think it's time to adjust the comment of `NewPublicKeyFromString`. To something like that, we need to specify that it accepts only compressed representation:
```
NewPublicKeyFromString returns a public key created from the given hex string public key representation in compressed form. 
```

Comment (AnnaShaleva) [reply pkg/crypto/keys/publickey.go:412]: May be done in a separate commit.

[the remark this answers is not in the document]
Comment (roman-khimov) [reply pkg/crypto/keys/publickey.go:412]: Easier to find as well.

File: cli/cmdargs/parser_test.go [hunk 2]
```diff
@@ -45,7 +44,7 @@ func TestParseCosigner(t *testing.T) {
 			Scopes:           transaction.CalledByEntry | transaction.CustomContracts,
 			AllowedContracts: []util.Uint160{c1, c2},
 		},
-		acc.StringLE() + ":CustomGroups:" + hex.EncodeToString(priv.PublicKey().Bytes()): {
+		acc.StringLE() + ":CustomGroups:" + priv.PublicKey().StringCompressed(): {
```
[the remark this answers is not in the document]
Comment (AnnaShaleva) [reply cli/cmdargs/parser_test.go:47]: Please, be more attentive, not resolved yet. Commit message is still the same:
```
Add CompressedHexString to get a string representation of the key in
compressed form.
```

File: pkg/crypto/keys/publickey.go [hunk 3]
```diff
@@ -406,3 +406,9 @@ func (p *PublicKey) UnmarshalYAML(unmarshal func(any) error) error {
 	}
 	return p.DecodeBytes(b)
 }
+
+// StringCompressed returns the hex string representation of the public key
+// in its compressed form.
+func (p *PublicKey) StringCompressed() string {
```
[the remark this answers is not in the document]
Comment (AnnaShaleva) [reply pkg/crypto/keys/publickey.go:412]: Please, be more attentive, not resolved.
```
	standByCommittee[0] = hex.EncodeToString(pubs[2].Bytes())
```
```
				o.Log.Debug("invalid response signature",
					zap.String("pub", hex.EncodeToString(pub.Bytes())))
```
```
				Signatures: map[string][]byte{
					hex.EncodeToString(priv.PublicKey().Bytes()): sign,
				},
```
```
	return it.Signatures[hex.EncodeToString(pub.Bytes())]
```
```
	pubHex := hex.EncodeToString(pub.Bytes())
```
```
			hex.EncodeToString(priv1.PublicKey().Bytes()): random.Bytes(keys.SignatureLen),
			hex.EncodeToString(priv2.PublicKey().Bytes()): random.Bytes(keys.SignatureLen),
```
```
PublicKey: hex.EncodeToString(g.PublicKey.Bytes()),
```
```
return json.Marshal(hex.EncodeToString(d.Group().Bytes()))
```

**Comment by AnnaShaleva** [comment]
Contribution guideline job is failing.

**Review by AnnaShaleva — APPROVED** [review]

**Pull request merged**